src/Composer/Util/RemoteFilesystem.php

```
@@ -299,7 +299,7 @@ protected function get($originUrl, $fileUrl, $additionalOptions = array(), $file
                 $e = new TransportException('Content-Length mismatch, received '.Platform::strlen($result).' bytes out of the expected '.$contentLength);
                 $e->setHeaders($http_response_header);
                 $e->setStatusCode($this->findStatusCode($http_response_header));
-                $e->setResponse($result);
+                $e->setResponse($this->decodeResult($result, $http_response_header));
```

I am not 100% sure whether we should attempt to decode for content-length mismatch

---

Yeah sounds like you rather not decode in this case, or try to decode and if it fails use the original, but not fail hard with a decode error as that just makes things worse :)